dashboard/app/helpers/levels_helper.rb

```diff
@@ -55,9 +55,6 @@ def get_channel_for(level, user = nil, make_readonly_with_other_user = true)
       # set_level_source to load answers when looking at another user,
       # we have to load the channel here.
       channel_token = ChannelToken.find_channel_token(level, user)
-      if make_readonly_with_other_user
-        readonly_view_options # TODO: has side effects
-      end
```

Comment:
get_channel_for is called in two places, the pair programming place (where we never want readonly) and the beginning of `app_options`, which is where we'll now do this check and potentially set readonly